### PR TITLE
Check whether loading is false before call onPress

### PR DESCRIPTION
**Pull request by imcvampire**
Implement for https://github.com/react-native-elements/react-native-elements/issues/2141

### CODE REVIEW

**Comment by imcvampire** [comment]
> Going to need add a test for this.
> 
> Basically the test should shallow render the button and pass in a stub onPress. After clicking on the button, assert that the stub hasn't been called

Added!

File: src/buttons/__tests__/Button.js [hunk 1]
```diff
@@ -43,6 +43,19 @@ describe('Button Component', () => {
     expect(onPress).toHaveBeenCalled();
   });
 
+  it('should not be call onPress events when loading is true', () => {
+    const onPress = jest.fn();
+    console.log = jest.fn();
+    const wrapper = shallow(<Button theme={theme} />);
+
+    wrapper.setProps({ onPress, loading: true });
```
Comment (iRoachie) [inline src/buttons/__tests__/Button.js]: No need to put this here just set it on the button directly. Also, remove the console.log mock

File: src/buttons/__tests__/Button.js [hunk 2]
```diff
@@ -43,6 +43,18 @@ describe('Button Component', () => {
     expect(onPress).toHaveBeenCalled();
   });
 
+  it('should not be call onPress events when loading is true', () => {
+    const onPress = jest.fn();
+    const wrapper = shallow(<Button theme={theme} loading={true} />);
+
+    wrapper.setProps({ onPress });
```
Comment (iRoachie) [inline src/buttons/__tests__/Button.js]: The onPress as well. Just it on the button.

The only time you'll need to use setProps is when you're testing before and after states.

File: src/buttons/__tests__/Button.js [hunk 3]
```diff
@@ -43,6 +43,18 @@ describe('Button Component', () => {
     expect(onPress).toHaveBeenCalled();
   });
 
+  it('should not be call onPress events when loading is true', () => {
+    const onPress = jest.fn();
+    const wrapper = shallow(<Button theme={theme} loading={true} />);
```
Comment (iRoachie) [inline src/buttons/__tests__/Button.js]:
```suggestion
    const wrapper = shallow(<Button theme={theme} loading />);
```

File: src/buttons/Button.js [hunk 4]
```diff
@@ -30,11 +30,18 @@ class Button extends Component {
...
+    if (!loading) {
+      onPress();
+    }
+  };
+
   render() {
     const {
       TouchableComponent,
       containerStyle,
-      onPress,
```
Comment (iRoachie) [inline src/buttons/Button.js]: This won't work. Removing onPress from here means it'll be in `...attributes`. Which means your onPress handler will be overridden and `handleOnPress` will never run

Comment (imcvampire) [reply src/buttons/Button.js]: I have corrected it

**Comment by iRoachie** [comment]
Thanks again 🚀

**Pull request merged**